File: app/src/main/java/org/tasks/injection/JobComponent.java
package org.tasks.injection;

import dagger.Subcomponent;
import org.tasks.jobs.AfterSaveWork;
import org.tasks.jobs.BackupWork;
import org.tasks.jobs.CleanupWork;
import org.tasks.jobs.DriveUploader;
import org.tasks.jobs.MidnightRefreshWork;
import org.tasks.jobs.RefreshWork;
import org.tasks.jobs.SyncWork;

@Subcomponent(modules = WorkModule.class)
public interface JobComponent {

  void inject(SyncWork syncWork);

  void inject(BackupWork backupWork);

  void inject(RefreshWork refreshWork);

  void inject(CleanupWork cleanupWork);

  void inject(MidnightRefreshWork midnightRefreshWork);

  void inject(AfterSaveWork afterSaveWork);

  void inject(DriveUploader driveUploader);
}

File: app/src/main/java/org/tasks/injection/JobComponent.kt
package org.tasks.injection

import dagger.Subcomponent
import org.tasks.jobs.*

@Subcomponent(modules = [WorkModule::class])
interface JobComponent {
    fun inject(work: SyncWork)
    fun inject(work: BackupWork)
    fun inject(work: RefreshWork)
    fun inject(work: CleanupWork)
    fun inject(work: MidnightRefreshWork)
    fun inject(work: AfterSaveWork)
    fun inject(work: DriveUploader)
    fun inject(work: ReverseGeocodeWork)
    fun inject(work: RemoteConfigWork)
}